pkg/resourcequotas/resourcequotas.go

```diff
@@ -0,0 +1,214 @@
+package resourcequotas
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1Typed "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+// Builder provides struct for resource quotas containing connection to the cluster.
+type Builder struct {
+	// Resource Quota definition
+	Definition *corev1.ResourceQuota
+
+	// Created resource quota object
+	Object *corev1.ResourceQuota
+
+	// Used in functions that define or mutate deployment definition. errorMsg is processed before the deployment
+	// object is created.
+	errorMsg  string
+	apiClient corev1Typed.CoreV1Interface
+}
+
+// NewBuilder creates a new resource quota builder.
+func NewBuilder(apiClient *clients.Settings, name, nsname string) *Builder {
+	glog.V(100).Infof("Initializing new resource quota structure with the following params: "+
+		"name=%s, namespace=%s", name, nsname)
+
+	builder := &Builder{
+		apiClient: apiClient.CoreV1Interface,
```

Comment:
I think we still need to set the definition with ObjectMeta, otherwise name and nsname aren't used and Definition ends up nil. It seems like Definition being non-nil is an invariant in the builders

Reply:
Whoops yes, I missed changing this to the struct definition.